src/components/CharacterForm/CharacterForm.js

```diff
@@ -54,17 +55,39 @@ const CharacterForm = ({ onClose }) => {
   const [errors, setErrors] = useState({})
   const { params } = router.query
 
+  const isEdit = useMemo(() => {
+    return params && params[0] === formModeEnum.edit
+  }, [params])
+
+  const character = useSelector((state) =>
+    isEdit ? getCharacterById(state, params[1]) : null
+  )
```

Comment:
This is not the ideal way of calling selector functions, please look into the concept of "currying" in javascript